### PR TITLE
feat: implemented mdbx trie cursor

### LINKED ISSUE #212: [MDBX] Implement trie cursor for historical proof storage
### Describe the feature

We need to implement a trie cursor interface that allows seeking and iterating over trie nodes stored in the historical proof database. This cursor will enable building proofs efficiently for both account and storage tries at any given block.

Goals:
- Implement `trie_cursor` method
- Implement the `ExternalTrieCursor` trait

Acceptance Criteria:
- Cursor supports iteration over account and storage trie nodes efficiently.
- `seek` and `seek_exact` correctly handle lexicographical ordering of trie paths.
- Cursor respects `max_block_number` to return the state as of a given block.
- Unit tests validate correct traversal, seek behavior, and handling of edge cases.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sadiq1971**
Closes #212

### CODE REVIEW

**Review by emhane — APPROVED** [review]
lgtm

File: crates/optimism/trie/src/db/cursor.rs [hunk 1]
```diff
@@ -1,33 +1,261 @@
...
+    /// Initializes new `BlockNumberVersionedCursor`
+    pub const fn new(cursor: Cursor, max_block_number: u64) -> Self {
+        Self { _table: PhantomData, cursor, max_block_number }
+    }
+
+    /// Resolve the latest version for `key` with `block_number` <= `max_block_number`.
+    /// Strategy:
+    /// - `seek_by_key_subkey(key, max)` gives first dup >= max.
+    ///   - if exactly == max → it's our latest
+    ///   - if > max → `prev_dup()` is latest < max (or None)
```
Comment (meyer9) [inline crates/optimism/trie/src/db/cursor.rs]: tiny suggestion

```suggestion
    ///   - if > max → `prev_dup()` is latest <= max (or None)
```

File: crates/storage/db-api/src/cursor.rs [hunk 2]
```diff
@@ -62,9 +62,15 @@ pub trait DbCursorRO<T: Table> {
 
 /// A read-only cursor over the dup table `T`.
 pub trait DbDupCursorRO<T: DupSort> {
+    /// Positions the cursor at the prev KV pair of the table, returning it.
+    fn prev_dup(&mut self) -> PairResult<T>;
+
```
[the remark this answers is not in the document]
Comment (emhane) [reply crates/storage/db-api/src/cursor.rs]: agree, it's not a priority now. in general prefer to upstream new logic, once it's fully e2e tested.

File: crates/storage/db/src/implementation/mdbx/cursor.rs [hunk 3]
```diff
@@ -158,11 +158,25 @@ impl<K: TransactionKind, T: Table> DbCursorRO<T> for Cursor<K, T> {
 }
 
 impl<K: TransactionKind, T: DupSort> DbDupCursorRO<T> for Cursor<K, T> {
+    /// Returns the previous `(key, value)` pair of a DUPSORT table.
```
Comment (emhane) [inline crates/storage/db/src/implementation/mdbx/cursor.rs]:
```suggestion
    /// Returns the previous kv-pair of a DUPSORT table.
```
just fyi, also usually written like this for simplicity

**Review by itschaindev — APPROVED** [review]
lgtm

File: crates/storage/db/src/implementation/mdbx/cursor.rs (hunk 3 above)
Comment (sadiq1971) [reply crates/storage/db/src/implementation/mdbx/cursor.rs]: Yep - nicer. Just followed existing comment style.
